File: src/main/java/com/pradera/poc/domain/enumeration/BlockType.java
package com.pradera.poc.domain.enumeration;

/**
 * The BlockType enumeration.
 */
public enum BlockType {
    TITLE,
    PARAGRAPH,
    CHAPTER;

    public static BlockType fromString(String editorType) {
        switch (editorType) {
            case "heading":
                return TITLE;
            case "paragraph":
                return PARAGRAPH;
            case "chapter":
                return CHAPTER;
        }
        throw new RuntimeException("unknown editor block type:" + editorType);
    }

    public String toEditorType() {
        switch (this) {
            case TITLE:
                return "heading";
            case PARAGRAPH:
                return "paragraph";
            case CHAPTER:
                return "chapter";
        }
        return "";
    }
}

File: src/main/java/com/pradera/poc/domain/enumeration/BlockType.kt
package com.pradera.poc.domain.enumeration

import java.lang.RuntimeException

/**
 * The BlockType enumeration.
 */
enum class BlockType {
    TITLE, PARAGRAPH, CHAPTER;

    fun toEditorType(): String {
        return when (this) {
            TITLE -> "heading"
            PARAGRAPH -> "paragraph"
            CHAPTER -> "chapter"
        }
    }

    companion object {
        @JvmStatic
        fun fromString(editorType: String): BlockType {
            return when (editorType) {
                "heading" -> TITLE
                "paragraph" -> PARAGRAPH
                "chapter" -> CHAPTER
                else -> null
            } ?: throw RuntimeException("unknown editor block type:$editorType")
        }
    }
}
